distributed_training/validator/reward.py

```diff
@@ -416,11 +422,15 @@ async def get_rewards(
             )
 
             # Final balanced score calculation with all_reduce
-            scores = blacklist_scores * (
-                (0.5 * gradient_scores * steps_scores) + (0.5 * all_reduce_scores)
+            scores = (
+                blacklist_scores
+                * ((0.5 * gradient_scores * steps_scores) + (0.5 * all_reduce_scores))[
```

Comment:
I think that might be too harsh given the nature of hivemind's all_reduce is that it doesn't work perfectly each step in allowing peers to join a group and sometimes even if they join the connection drops out.

Reply:
The average of a multiplicative approach though might end up favouring honest miners over malicious ones so it could end up being a better reward mechanism. We'd need to test this to be sure.